Add audit logs for session management

ShanChathusanda93 opened the pull request:
### Proposed changes in this pull request

* Added the audit logs for the session management.
* Session add and session terminate with session ID and terminate all the session for a user are now have audit logs.

senthalan commented on components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:
```diff
@@ -150,6 +153,8 @@ private void terminateSessionsOfUser(List<String> sessionList) {
 
         for (String session : sessionList) {
             sessionManagementService.removeSession(session);
+            addAuditLogs(session, FrameworkUtils.getLoginTenantDomainFromContext(),
```
for terminate audit logs, shall we add to https://github.com/wso2/carbon-identity-framework/blob/50552a805f2b03c7ffe8457469376b0dd0fc6f0a/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/ServerSessionManagementServiceImpl.java#L58 ?

senthalan commented:
let's also check session extending(using the API call and passive request) and session update(SSO with a new application) events and check whether we have audit events for these.

ShanChathusanda93 commented:
Checked operations list
1. Session Extend via API call : Audit data available.
2. Session extend via Passive call : 
3. Session update via SSO : Introduced audit logs are available.
4. Password reset to check the session terminate audit : Introduced audit logs are available

ShanChathusanda93 commented on components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:
```diff
@@ -1029,4 +1034,16 @@ private void setAuthenticatedIDPsOfApp(SessionContext sessionContext,
+        auditData.put(SessionMgtConstants.SESSION_CONTEXT_ID, sessionKey);
+        auditData.put(SessionMgtConstants.TENANT_DOMAIN, tenantDomain);
+        auditData.put(SessionMgtConstants.TRACE_ID, traceId);
+        auditData.put(SessionMgtConstants.SESSION_CREATED_TIMESTAMP, createdTimestamp);
```
As we discussed offline, at this point we cannot add the expiry time because the expiry time is calculated from a following step. Hense added the remember me flag to the audit log.
